components/brave_wallet_ui/page/async/wallet_page_async_handler.ts

```diff
@@ -44,6 +46,41 @@ async function refreshWalletInfo (store: Store) {
   store.dispatch(WalletActions.initialized({ ...result, selectedAccount: '', visibleTokens: [] }))
 }
 
+async function importFromExternalWallet (
+  walletType: BraveWallet.ExternalWalletType,
+  payload: ImportFromExternalWalletPayloadType
+): Promise<ImportWalletErrorPayloadType> {
+  const {
+    braveWalletService,
+    keyringService
+  } = getWalletPageApiProxy()
+
+  const result = await braveWalletService.importFromExternalWallet(
+    walletType,
+    payload.password,
+    payload.newPassword
+  )
+
+  // complete backup if a new password was provided
+  if (payload.newPassword && result.success) {
+    keyringService.notifyWalletBackupComplete()
+  }
+
+  // was the provided import password correct?
+  const checkExistingPasswordError = result.errorMessage === getLocale('braveWalletImportPasswordError')
+    ? result.errorMessage
+    : undefined
+
+  // was import successful (if attempted)
+  const importError = payload.newPassword ? result.errorMessage || undefined : undefined
+
+  return {
+    hasError: !!(importError || checkExistingPasswordError),
+    errorMessage: importError || checkExistingPasswordError,
+    incrementAttempts: true
+  }
+}
```

Comment:
Updated wallet import implementation so that we can check the import password before completing the import process